token/core/fabtoken/ppm/ppm.go

```diff
@@ -47,21 +37,28 @@ type PublicParamsManager struct {
 }
 
 // NewPublicParamsManager initializes a PublicParamsManager with the passed PublicParamsLoader
-func NewPublicParamsManager(PPLabel string, vault Vault, publicParamsLoader PublicParamsLoader) *PublicParamsManager {
-	return &PublicParamsManager{PPLabel: PPLabel, Vault: vault, PublicParamsLoader: publicParamsLoader, Mutex: sync.RWMutex{}}
+func NewPublicParamsManager(PPLabel string, vault Vault) *PublicParamsManager {
+	return &PublicParamsManager{
+		PPLabel: PPLabel,
+		Vault:   vault,
+	}
 }
 
 // NewPublicParamsManagerFromParams initializes a PublicParamsManager with the passed PublicParams
 func NewPublicParamsManagerFromParams(pp *fabtoken.PublicParams) (*PublicParamsManager, error) {
 	if pp == nil {
 		return nil, errors.Errorf("public parameters must be non-nil")
 	}
-	return &PublicParamsManager{PP: pp, Mutex: sync.RWMutex{}}, nil
+	return &PublicParamsManager{PP: pp, PPLabel: pp.Label}, nil
 }
 
 // PublicParameters returns the public parameters of PublicParamsManager
 func (v *PublicParamsManager) PublicParameters() driver.PublicParameters {
-	return v.PublicParams()
+	pp := v.PublicParams()
+	if pp == nil {
```

Comment:
What is the difference?

Reply:
pp is a struct, if pp is nil and return as an interface, what you get is a pointer to a nil struct which is not nil.